app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/notification/DeviceShieldReminderNotificationScheduler.kt

```diff
@@ -27,39 +27,48 @@ import androidx.work.PeriodicWorkRequestBuilder
 import androidx.work.WorkManager
 import com.duckduckgo.app.di.AppCoroutineScope
 import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.app.global.plugins.PluginPoint
 import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.vpn.AppTpVpnFeature
+import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
 import com.duckduckgo.mobile.android.vpn.feature.removal.VpnFeatureRemover
 import com.duckduckgo.mobile.android.vpn.service.TrackerBlockingVpnService
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.Type.DISABLED
+import com.duckduckgo.mobile.android.vpn.service.VpnStaticNotificationContentPlugin.Type.REVOKED
 import com.duckduckgo.mobile.android.vpn.service.VpnReminderNotificationWorker
 import com.duckduckgo.mobile.android.vpn.service.VpnReminderReceiver
 import com.duckduckgo.mobile.android.vpn.service.VpnServiceCallbacks
+import com.duckduckgo.mobile.android.vpn.service.getHighestPriorityPluginForType
 import com.duckduckgo.mobile.android.vpn.state.VpnStateMonitor.VpnStopReason
 import com.squareup.anvil.annotations.ContributesMultibinding
-import dagger.SingleInstanceIn
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 import logcat.logcat
 
-@SingleInstanceIn(VpnScope::class)
 @ContributesMultibinding(VpnScope::class)
 class DeviceShieldReminderNotificationScheduler @Inject constructor(
     private val context: Context,
     private val workManager: WorkManager,
     private val notificationManager: NotificationManagerCompat,
-    private val deviceShieldAlertNotificationBuilder: DeviceShieldAlertNotificationBuilder,
     private val vpnFeatureRemover: VpnFeatureRemover,
     private val dispatchers: DispatcherProvider,
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
+    private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+    private val vpnStaticNotificationContentPluginPoint: PluginPoint<VpnStaticNotificationContentPlugin>,
 ) : VpnServiceCallbacks {
 
     override fun onVpnStarted(coroutineScope: CoroutineScope) {
-        scheduleUndesiredStopReminderAlarm()
-        cancelReminderForTomorrow()
-        hideReminderNotification()
-        enableReminderReceiver()
+        if (vpnFeaturesRegistry.isFeatureRegistered(AppTpVpnFeature.APPTP_VPN)) {
```

Comment:
Why do we need to check AppTP feature specifically if we added the plugin point?

Reply:
We probably don't need to. Ill check this tom.

Reply:
@aitorvs SO I checked this, this part of the code doesn't access any content specifically. It just schedules them.

I think I can remove this since:
- scheduleUndesiredStopReminderAlarm() - The worker schedule here checks if there's an active content plugin anyway.
- cancelReminderForTomorrow() - should work regardless of what feature is started.
- hideReminderNotification() - should work regardless of what feature is started.
-  enableReminderReceiver() - should work regardless of what feature is started.